### PR TITLE
feat: do not update/commit files in .gitignore

### PR DESCRIPTION
**Pull request by bcoe**
I've bumped into this issue a few times, and noticed a similar conversation in #213.

How does this solution look? @nexdrew, @satazor, @arobson?

### CODE REVIEW

**Comment by satazor** [comment]
Looks good, though tests are failing

File: lib/gitignore.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const Dotignore = require('@bcoe/dotignore')
```
Comment (nexdrew) [inline lib/gitignore.js]: For Node 4 support, this file needs `'use strict'` at the top. This fixes the failing tests for me locally.

**Comment by nexdrew** [comment]
My attempt to fix the Windows build failed. Anyone else have ideas?

File: lib/gitignore.js (hunk 1 above)
Comment (bcoe) [reply lib/gitignore.js]: @nexdrew I guess I'll need to spin up a Windows VM ... grumble.

**Comment by bcoe** [comment]
@nexdrew I noticed the underlying library has some logic that uses `/` rather than `path.sep`; I might take @stevemao's advice and write our own little library that uses `find-up` and will patch the issue.

**Comment by nexdrew** [comment]
@bcoe Are you talking about [this line](https://github.com/bcoe/dotignore/blob/90442b56563695be48bfced806faffc63e44ac4e/index.js#L29) (and maybe line 9 as well)? I don't really think this would affect the failing test, but we could try setting `matcher. delimiter = path.sep`.

**Comment by nexdrew** [comment]
@bcoe I was able to spin up a Windows VM, and it looks like my suggestion of `matcher.delimiter = path.sep` actually works/fixes the failing test (which I find a bit surprising). Not sure my suggestion is the best long-term solution, but it might be enough to get this PR over the finish line.

File: lib/gitignore.js (hunk 1 above)
Comment (bcoe) [reply lib/gitignore.js]: @nexdrew smart 👍

**Comment by Aigeec** [comment]
Hey guys just wanted to let you know that I raised a bug on (dotgitignore)[https://github.com/bcoe/dotgitignore/issues/1]. It means that lines that are comments are included in the matches. So for example someone puts their directory name is as a comment the `dotgitignore` starts matching everything.

**Comment by chrisgalvan** [comment]
@Aigeec Any update on when that will be fixed I just bump into that situation where my package.json and package-lock.json where ignored because some weird issue with https://github.com/bcoe/dotgitignore where it does match some ignored lines withe the full path of the package.json.

For example my package.json is at:
c:/user/myuser/dev/mypackage/package.json

and in my .gitignore I have the following
# ignore user files
.user*

because the path contains `user` package.json is being ignored.

**Comment by Aigeec** [comment]
I've open a PR https://github.com/bcoe/dotgitignore/pull/3 on https://github.com/bcoe/dotgitignore to resolve the issue but I believe @bcoe is triaging it at the moment.

**Pull request merged**